tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -570,6 +572,44 @@ public function testAlterTableScenario()
         }
     }
 
+
+    public function testAlterTableInNamespace()
+    {
+        if (! $this->_sm->getDatabasePlatform()->supportsSchemas()) {
+            $this->markTestSkipped('Schema definition is not supported by this platform.');
+        }
+
+        if (! in_array('testschema', $this->_sm->listNamespaceNames(), true)) {
```

Comment:
Instead of checking if the schema already exists, could the test create the schema unconditionally and drop it on teardown? Otherwise, if a previous test execution didn't exit cleanly (e.g. the schema was created but the table wasn't), the subsequent executions will always fail.

Reply:
I trusted on the fact that the database is dropped on the beginning of the test. This would remove the schema, too. However, I'll try to implement it more specific.